src/main/java/seedu/address/model/scheduler/Scheduler.java

```diff
@@ -0,0 +1,124 @@
+package seedu.address.model.scheduler;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.parser.IndexHandler;
+import seedu.address.model.Model;
+import seedu.address.model.person.ContactIndex;
+import seedu.address.model.person.Person;
+import seedu.address.model.timetable.SchoolDay;
+import seedu.address.model.timetable.TimePeriod;
+import seedu.address.model.timetable.TimeSlot;
+import seedu.address.model.timetable.Timetable;
+
+/**
+ * Represents an automatic scheduler.
+ */
+public class Scheduler {
+    private List<Timetable> schedules;
+    private Model model;
+    private List<Person> participants;
+
+    /**
+     * Constructs a scheduler.
+     * @param model
+     */
+    public Scheduler(Model model) {
+        this.model = model;
+        this.schedules = new ArrayList<>();
+        this.participants = new ArrayList<>();
+    }
+
+    /**
+     * Initialises the scheduler with all the participants.
+     * @param participantIndices
+     */
+    public void initialise(List<ContactIndex> participantIndices) {
```

Comment:
Could this be a polymorphised constructor?

Reply:
Please do elaborate what do you mean here? Thanks!

Reply:
Scheduler could have another constructor accepting `List<ContactIndex>`.